### ISSUE TITLE
Add more workflow information for Github workflow certificates (sha, run)

### ISSUE THREAD

**Issue by asraa** [sigstore/fulcio#208]
**Description**

It would be good to have information on the sha that the builder is requesting a certificate for in the case of a github workflow. this would allow a user to authenticate when the workflow was running, and trust that a builder is signing on a specific sha (if branch protection is off at some point in a repo, anyone can make a malicious signed artifact with a builder in their malicious branch)

This will help getting repos up to SLSA 2!

Related:
https://github.com/sigstore/fulcio/pull/204/files

**Comment by laurentsimon** [comment]
Would it be useful to have a provider-agnostic extension?
Say we have Google and GitHub providers. Depending on context, we may want to add additional information (in the case of this issue, a commit hash): will the OID have a {provider type, datablob} structure?
I don't think we want to create a new extension every time a provider needs to add additional context to the cert.
wdut?

**Comment by asraa** [comment]
cc @bobcallaway

**Comment by asraa** [comment]
Or barring privacy concerns around logging the full github token in rekor (are there any? maybe private workflows?) maybe we can serialize the whole token and stick it in a single extension?

**Comment by laurentsimon** [comment]
yeah that's what I was thinking too: `version || len_provider_name || provider_name || provider_version || serialized_data`. `provider_version` not necessary.

**Comment by asraa** [comment]
Side note: @dlorenc got the same effect as this by signing with annotations on the sha/commit.

I *think* it's still useful to have this information in the cert in case it's not a cosign container sig being created (just a signature on an artifact, and this info can come from the cert)

**Comment by haydentherapper** [comment]
@asraa Good to close this issue as fixed?
